pkg/network/tcp_peer.go

```diff
@@ -270,15 +272,20 @@ func (p *TCPPeer) Handshaked() bool {
 	return p.handShake == (verAckReceived | verAckSent | versionReceived | versionSent)
 }
 
+// IsFullNode returns whether the node has full capability or TCP/WS only
+func (p *TCPPeer) IsFullNode() bool {
+	return p.Handshaked() && p.isFullNode
```

Comment:
Either `isFullNode` should be atomic or it should be accessed with a lock held.

Reply:
Can we use both lock and `Handshaked` together?

Reply:
You can create an unlocked internal `handshaked` version of `Handshake`.